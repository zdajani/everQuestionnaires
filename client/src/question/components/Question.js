import React from 'react';
import PropTypes from 'prop-types';

const Question = ({ name, label }) => (
  <div>
    <h4>
      {name}
    </h4>
    <h6>
      {label}
    </h6>
  </div>  
);

Question.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};

export default Question;