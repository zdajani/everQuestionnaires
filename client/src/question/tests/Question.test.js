import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Question from '../components/Question';
import { formattedData } from './testData';

const setup = (question = formattedData) => {
  const component = shallow(
    <Question {...question}  />
  );

  return {
    component
  };
}

describe('Question component', () => {
  it('renders self with question props ', () => {
    const { component } = setup();
    expect(component.instance().props).toEqual({ 
      name: formattedData.name, 
      label:formattedData.label,
      id:  formattedData.id
    }); 
    
    expect(toJson(component)).toMatchSnapshot();
  });
});
