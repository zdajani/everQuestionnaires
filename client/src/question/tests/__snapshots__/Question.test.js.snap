// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component renders self with question props  1`] = `
<div>
  <h4>
    Ragdoll traits
  </h4>
  <h6>
    describe ragdoll traits
  </h6>
</div>
`;
